lib/molinillo/resolution.rb

```diff
@@ -266,14 +303,17 @@ def unwind_details_for_conflict(conflict, existing_unwind_details = nil)
         unwind_details = existing_unwind_details || UnwindDetails.new(-1, nil, nil, [])
 
         binding_requirements = binding_requirements_for_conflict(conflict)
+        trees = []
         binding_requirements.reverse_each do |r|
+          partial_tree = [r]
+          trees << partial_tree
           # If this requirement has alternative possibilities, check if any would
           # satisfy the other requirements that created this conflict
           requirement_state = find_state_for(r)
           candidate_index = states.index(requirement_state)
-          next if candidate_index && candidate_index < unwind_details.state_index
+          next if candidate_index && UnwindDetails.new(candidate_index, r, partial_tree) < unwind_details
```

Comment:
can we use one temporary `UnwindDetails` instead of creating a new one inside the loop?